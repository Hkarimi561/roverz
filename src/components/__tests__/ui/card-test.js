/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '@ui/Card';

it('Card renders correctly', () => {
  const tree = renderer.create(
    <Card><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Card w/ Title renders correctly', () => {
  const tree = renderer.create(
    <Card title={'With Title'}><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Containerstyle renders correctly', () => {
  const tree = renderer.create(
    <Card containerStyle={{ backgroundColor: 'red', borderRadius: 5, borderColor: 'red', borderWidth: 2 }}>
      <Text>Hello world</Text></Card>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

/* eslint-disable global-require */
it('Card w/ Image renders correctly', () => {
  const img = require('../../../images/image.png');// eslint-disable-line global-require
  const tree = renderer.create(
    <Card image={img}><Text>Hello world</Text></Card>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
/* eslint-enable global-require */

// describe('containerstyle', () => {
//   it('should match containerstyle ', () => {
//     const containerstyle = { backgroundColor: 'red', borderRadius: 5, borderColor: 'red', borderWidth: 2 };
//     expect(containerstyle).toBe(Card.props.containerStyle);
//     // expect(setAvType('direct')).toBe('perm-identity');
//     // expect(setAvType('')).toBe('supervisor-account');
//   });
// });
