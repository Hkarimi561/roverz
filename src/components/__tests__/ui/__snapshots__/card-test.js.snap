// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  0={
    Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Hello world
    </Text>,
      "containerStyle": Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
          "borderWidth": 0,
        },
      ],
      "dividerStyle": Array [
        Object {
          "backgroundColor": "#D0D1D5",
        },
      ],
      "titleStyle": Array [
        Object {
          "color": "#373856",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 27,
          "fontWeight": "800",
          "left": 0,
          "lineHeight": 34,
          "margin": 0,
          "marginBottom": 4,
          "right": 0,
        },
        Object {
          "marginBottom": 15,
        },
      ],
    }
  }
  1={Array []}
  2={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Card w/ Image renders correctly 1`] = `
<View
  0={
    Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Hello world
    </Text>,
      "containerStyle": Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
          "borderWidth": 0,
        },
      ],
      "dividerStyle": Array [
        Object {
          "backgroundColor": "#D0D1D5",
        },
      ],
      "image": 1,
      "titleStyle": Array [
        Object {
          "color": "#373856",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 27,
          "fontWeight": "800",
          "left": 0,
          "lineHeight": 34,
          "margin": 0,
          "marginBottom": 4,
          "right": 0,
        },
        Object {
          "marginBottom": 15,
        },
      ],
    }
  }
  1={Array []}
  2={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Card w/ Title renders correctly 1`] = `
<View
  0={
    Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Hello world
    </Text>,
      "containerStyle": Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
          "borderWidth": 0,
        },
      ],
      "dividerStyle": Array [
        Object {
          "backgroundColor": "#D0D1D5",
        },
      ],
      "title": "With Title",
      "titleStyle": Array [
        Object {
          "color": "#373856",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 27,
          "fontWeight": "800",
          "left": 0,
          "lineHeight": 34,
          "margin": 0,
          "marginBottom": 4,
          "right": 0,
        },
        Object {
          "marginBottom": 15,
        },
      ],
    }
  }
  1={Array []}
  2={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`Containerstyle renders correctly 1`] = `
<View
  0={
    Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
    >
        Hello world
    </Text>,
      "containerStyle": Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D0D1D5",
          "borderRadius": 2,
          "borderWidth": 0,
        },
      ],
      "dividerStyle": Array [
        Object {
          "backgroundColor": "#D0D1D5",
        },
      ],
      "titleStyle": Array [
        Object {
          "color": "#373856",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 27,
          "fontWeight": "800",
          "left": 0,
          "lineHeight": 34,
          "margin": 0,
          "marginBottom": 4,
          "right": 0,
        },
        Object {
          "marginBottom": 15,
        },
      ],
    }
  }
  1={
    Object {
      "backgroundColor": "red",
      "borderColor": "red",
      "borderRadius": 5,
      "borderWidth": 2,
    }
  }
  2={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
